src/server/api/routers/university.ts

```diff
@@ -0,0 +1,66 @@
+import { z } from "zod";
+import { createTRPCRouter, publicProcedure } from "@/server/api/trpc";
+
+export const universityRouter = createTRPCRouter({
+  createUniversity: publicProcedure
+    .input(
+      z.object({ name: z.enum(["SMU", "NTU", "NUS"]), siteUrl: z.string() }),
+    )
+    .query(async ({ input, ctx }) => {
+      const university = await ctx.db.universities.create({
+        data: {
+          name: input.name,
+          siteUrl: input.siteUrl,
+        },
+      });
+      return university;
+    }),
```

Comment:
**suggestion:** use `.mutation(...)` instead of `.query(...)` for procedure calls that creates/updates/deletes

rationale:

> Unlike queries, mutations are typically used to create/update/delete data or perform server side-effects. For this purpose, TanStack Query exports a useMutation hook.
>
> source \- [`react-query` docs on mutations](https://tanstack.com/query/v4/docs/react/guides/mutations) (`trpc` uses a thin wrapper over `react-query`)

alternative community discussions @ trpc:

> Main reasons:
> - semantics -- it's a lot easier to reason about your codebase with the distinction
> - queries happens through HTTP GET --> easy HTTP caching
> - mutations happens through HTTP POST --> not cached
> 
> source \- `https://github.com/trpc/trpc/discussions/1638`

```suggestion
  createUniversity: publicProcedure
    .input(
      z.object({ name: z.enum(["SMU", "NTU", "NUS"]), siteUrl: z.string() }),
    )
    .mutation(async ({ input, ctx }) => {
      const university = await ctx.db.universities.create({
        data: {
          name: input.name,
          siteUrl: input.siteUrl,
        },
      });
      return university;
    }),
```

Reply:
Thanks for the insights! I have implemented the change at [9468d37](https://github.com/AfterClass-io/afterclass.io-v2/pull/39/commits/9468d379acbd59fe877cb1ed00f3a6a6e6964374) and [384f435](https://github.com/AfterClass-io/afterclass.io-v2/pull/39/commits/384f43535439631f04db20a031f3cd5492cd1f6b)